lightning/src/ln/channel.rs

```diff
@@ -5446,13 +5446,15 @@ impl<SP: Deref> FundedChannel<SP> where
 						is_splice: true,
 					})
 				} else {
-					Err("Channel is not actively refunding")
+					Err("Received unexpected interactive transaction negotiation message: \
+						the channel is splicing, but splice_init/splice_ack has not been exchanged yet")
 				}
 			} else {
-				Err("Channel is not refunding")
+				Err("Received unexpected interactive transaction negotiation message: \
+					the channel is splicing, but splice_init/splice_ack has not been exchanged yet")
```

Comment:
Could we make these distinct messages? Seems like the first is saying we are waiting for one of those messages. But shouldn't the second say we've already received those messages and are now waiting on `splice_locked`?

Reply:
No, these are the same error case, but it's two if's as it's an Option within an Option.

Reply:
Don't we need to move `PendingSplice::funding_scope` into `FundedChannel::pending_funding` once `tx_signatures` have been exchanged? This would leave the former as `None` and the latter as `Some`. And while waiting on confirmations (in order to exchange `splice_locked`) we cannot start a new splice (only an RBF).